internal/core/algorithm/ngt/option.go

```diff
@@ -104,58 +115,69 @@ func WithDistanceTypeByString(dt string) Option {
 func WithDistanceType(t distanceType) Option {
 	return func(n *ngt) error {
 		switch t {
 		case L1:
 			if C.ngt_set_property_distance_type_l1(n.prop, n.ebuf) == ErrorCode {
-				return errors.ErrFailedToSetDistanceType(n.newGoError(n.ebuf), "L1")
+				err := errors.ErrFailedToSetDistanceType(n.newGoError(n.ebuf), "L1")
+				return errors.NewErrCriticalOption("distanceType", t, err)
 			}
 		case L2:
 			if C.ngt_set_property_distance_type_l2(n.prop, n.ebuf) == ErrorCode {
-				return errors.ErrFailedToSetDistanceType(n.newGoError(n.ebuf), "L2")
+				err := errors.ErrFailedToSetDistanceType(n.newGoError(n.ebuf), "L2")
+				return errors.NewErrCriticalOption("distanceType", t, err)
 			}
 		case Angle:
 			if C.ngt_set_property_distance_type_angle(n.prop, n.ebuf) == ErrorCode {
-				return errors.ErrFailedToSetDistanceType(n.newGoError(n.ebuf), "Angle")
+				err := errors.ErrFailedToSetDistanceType(n.newGoError(n.ebuf), "Angle")
+				return errors.NewErrCriticalOption("distanceType", t, err)
 			}
 		case Hamming:
 			if C.ngt_set_property_distance_type_hamming(n.prop, n.ebuf) == ErrorCode {
-				return errors.ErrFailedToSetDistanceType(n.newGoError(n.ebuf), "Hamming")
+				err := errors.ErrFailedToSetDistanceType(n.newGoError(n.ebuf), "Hamming")
+				return errors.NewErrCriticalOption("distanceType", t, err)
 			}
 		case Cosine:
 			if C.ngt_set_property_distance_type_cosine(n.prop, n.ebuf) == ErrorCode {
-				return errors.ErrFailedToSetDistanceType(n.newGoError(n.ebuf), "Cosine")
+				err := errors.ErrFailedToSetDistanceType(n.newGoError(n.ebuf), "Cosine")
+				return errors.NewErrCriticalOption("distanceType", t, err)
 			}
 		case NormalizedAngle:
 			if C.ngt_set_property_distance_type_normalized_angle(n.prop, n.ebuf) == ErrorCode {
-				return errors.ErrFailedToSetDistanceType(n.newGoError(n.ebuf), "NormalizedAngle")
+				err := errors.ErrFailedToSetDistanceType(n.newGoError(n.ebuf), "NormalizedAngle")
+				return errors.NewErrCriticalOption("distanceType", t, err)
 			}
 		case NormalizedCosine:
 			if C.ngt_set_property_distance_type_normalized_cosine(n.prop, n.ebuf) == ErrorCode {
-				return errors.ErrFailedToSetDistanceType(n.newGoError(n.ebuf), "NormalizedCosine")
+				err := errors.ErrFailedToSetDistanceType(n.newGoError(n.ebuf), "NormalizedCosine")
+				return errors.NewErrCriticalOption("distanceType", t, err)
 			}
 		case Jaccard:
 			if C.ngt_set_property_distance_type_jaccard(n.prop, n.ebuf) == ErrorCode {
-				return errors.ErrFailedToSetDistanceType(n.newGoError(n.ebuf), "Jaccard")
+				err := errors.ErrFailedToSetDistanceType(n.newGoError(n.ebuf), "Jaccard")
+				return errors.NewErrCriticalOption("distanceType", t, err)
 			}
 		default:
-			return errors.ErrUnsupportedDistanceType
+			err := errors.ErrUnsupportedDistanceType
+			return errors.NewErrCriticalOption("distanceType", t, err)
 		}
 		return nil
 	}
 }
 
+// WithObjectTypeByString represents the option to set the object type for NGT.
```

Comment:
```suggestion
// WithObjectTypeByString represents the option to set the object type by string for NGT.
```
It would be better differ from the explanation of `WithObjectType` 🤔 
L192